Repositories/ProductMongoDBRepository.cs

```diff
@@ -0,0 +1,100 @@
+﻿using MongoDB.Driver;
+using InventoryManagement.Models;
+using System.Collections.Generic;
+using MongoDB.Bson;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using System.IO;
+namespace InventoryManagement.Repositories
+{
+    public class ProductMongoDBRepository : IProductRepository
+    {
+        private IMongoClient _client;
+        private IMongoCollection<BsonDocument> _collection;
+        private IMongoDatabase _database;
+
+
+        public void SetupDB()
+        {
+            var config = new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile("appsettings.json").Build();
+            string connectionStrong = config["MongoDB:connectionString"];
+
+            try
+            {
+                _client = new MongoClient(connectionStrong);
+                _database = _client.GetDatabase("inventory");
+                _collection = _database.GetCollection<BsonDocument>("Inventory");
+
+            }
+            catch { throw; }
+
+        }
+
+        public IEnumerable<Product> GetProducts()
+        {
+            var products = new List<Product>();
+            var documents = _collection.Find(new BsonDocument()).ToList();
+            foreach (var document in documents)
+            {
+                string productName = (string)document.GetValue(1);
+                int productQuantity = (int)document.GetValue(2);
+                double productPrice = (double)document.GetValue(3);
+                Product product = new(productName, productQuantity, productPrice);
+                products.Add(product);
+
+            }
+            return products;
+        }
+
+        public void InsertProduct(Product product)
+        {
+            var document = new BsonDocument
+        {
+          { "productName", product.Name },
+          { "productQuantity", product.Quantity },
+          { "productPrice", product.Price }};
+            _collection.InsertOne(document);
+        }
+
+        public void UpdateProduct(string productName, Product product)
+        {
+            var filter = Builders<BsonDocument>.Filter.Eq("productName", productName);
+            var update = Builders<BsonDocument>.Update.Set("productName", product.Name).Set("productQuantity", product.Quantity).Set("productPrice", product.Price); ;
+
+            try
+            {
+                _collection.UpdateOne(filter, update);
+            }
+            catch { throw; }
+
+
+        }
+
+        public Product? RetrieveProductByName(string productName)
+        {
+
+
+            var filter = Builders<BsonDocument>.Filter.Eq("productName", productName);
+            var document = _collection.Find(filter).First();
+            if (document != null)
+            {
+                string name = document.GetValue("productName").AsString;
+                int quantity = document.GetValue("productQuantity").AsInt32;
+                double price = document.GetValue("productPrice").AsDouble;
```

Comment:
that would be another advantage of using strongly typed collections, that you don't have to parse the fields.